lib/web_ui/lib/src/engine/semantics/heading.dart

```diff
@@ -0,0 +1,23 @@
+// Copyright 2013 The Flutter Authors. All rights reserved.
+// Use of this source code is governed by a BSD-style license that can be
+// found in the LICENSE file.
+
+import '../dom.dart';
+import 'semantics.dart';
+
+/// Renders semantics objects as headings with the corresponding
+/// level (h1 ... h6).
+class Heading extends PrimaryRoleManager {
+  Heading(SemanticsObject semanticsObject)
+      : super.withBasics(PrimaryRole.heading, semanticsObject);
+
+  @override
+  void update() {
+    super.update();
+
+      if (semanticsObject.headingLevel != -1) {
+        semanticsObject.setAriaRole('heading');
```

Comment:
I think we should set the aria role in the constructor. It is expected that the primary role of a node never changes.